### DISCUSSION TITLE
Sync implementation for boxed queries

### DISCUSSION THREAD

**Discussion by dullbananas** [diesel-rs/diesel#3859] [Ideas]
Boxed queries should implement Sync (not just Send) so that a custom query can contain a reference to it. Because I can't do that, I instead need to make a query with a `Drop` implementation that moves the subquery into a cloned `Arc<Mutex<T>>` so that it can be accessed and used again.

https://docs.rs/diesele/0.1.0/src/diesele/lib.rs.html#302

**Comment by weiznich** [comment]
It's unfortunately not possible to change this without a breaking change. As we don't plan any new major diesel version bump anytime soon this cannot be implemented as far as I'm aware.
